needle.pm

```diff
@@ -276,13 +276,34 @@ sub has_tag {
 }
 
 sub has_property {
-    my ($self, $tag) = @_;
-    for my $t (@{$self->{properties}}) {
-        return 1 if ($t eq $tag);
+    my ($self, $property_name) = @_;
+    for my $property (@{$self->{properties}}) {
+        if (ref($property) eq "HASH") {
+            return 1 if ($property->{name} eq $property_name);
+        }
+        else {
+            return 1 if ($property eq $property_name);
+        }
     }
     return 0;
 }
 
+sub get_property_value {
+    my ($self, $property_name) = @_;
+
+    for my $property (@{$self->{properties}}) {
+        if (ref($property) eq "HASH") {
+            return $property->{value} if ($property->{name} eq $property_name);
+        }
+    }
+    if ($property_name eq "workaround") {
+        if ($self->{name} =~ /\S+\-(bsc|poo)(\d+)\-\S+/) {
```

Comment:
hm, this will not be long-term scalable but IIUC this is merely backward compatibility support, right?

Reply:
Have you seen this comment?

Reply:
Sorry, I forgot to reply.
Yes, you are right, This is to support the existed needles those have workaround property and the related ticket number was recorded in needle name. Maybe I should add a array to store those ticket head, such as bsc, poo and so on(I did not find out other ticket head in existed needles.)